test/e2e-common.sh

```diff
@@ -185,16 +185,25 @@ function install_istio() {
   kubectl apply -f "${INSTALL_ISTIO_YAML}" || return 1
   UNINSTALL_LIST+=( "${INSTALL_ISTIO_YAML}" )
 
+  PILOT_NAME="istiod"
+  if [[ "${ISTIO_VERSION}" == "1.4-latest" ]]; then
+    PILOT_NAME="istio-pilot"
+  fi
+
   echo ">> Patching Istio"
   # There are reports of Envoy failing (503) when istio-pilot is overloaded.
   # We generously add more pilot instances here to reduce flakes.
-  if kubectl get hpa -n istio-system istio-pilot 2>/dev/null; then
-    kubectl patch hpa -n istio-system istio-pilot \
+  if kubectl get hpa -n istio-system ${PILOT_NAME} 2>/dev/null; then
```

Comment:
Let's pull into the `istio*-ci*.yaml` template, since Markus already separate out a `lean` template better suite for development and these `ci` templates are exclusively for CI.